synapse/tests/test_lib_ast.py

```diff
@@ -823,6 +823,30 @@ async def test_ast_pivot(self):
             self.len(0, await core.nodes('it:host +inet:fqdn:zone'))
             self.len(1, await core.nodes('.created +inet:fqdn:zone=vertex.link'))
 
+            self.len(4, await core.nodes('[ inet:ip=1.2.3.4/30 ]'))
+
+            self.len(1, await core.nodes('[ it:network=* :net=1.2.3.4-1.2.3.7 ]'))
+
+            self.len(4, await core.nodes('it:network :net -> *'))
+            self.len(4, await core.nodes('it:network :net -> inet:ip'))
+
+            self.len(1, await core.nodes('[ test:str=foo :cidr=1.2.3.4/30 ]'))
+
+            self.len(5, await core.nodes('test:str=foo :cidr -> *'))
+            self.len(4, await core.nodes('test:str=foo :cidr -> inet:ip'))
+
+            # inet:cidr ends up pivoting to broadcast/network IPs twice
```

Comment:
Note: dedupe these

Reply:
These are now deduped by skipping props with a type that there is a special pivot defined for.